### ISSUE TITLE
Incorrect validation accuracy 

### ISSUE THREAD

**Issue by fatahfattah** [keras-team/keras#12689]
Currently working on a fashion dataset where I am predicting the clothing categories (e.g shoes, jeans, jackets etc.) My training/validation split is 2500/500, equally distributed over all the classes.

Accuracy after 20 epochs:
Train accuracy:
Final: train_loss: 0.1132 - train_acc: 0.9623

My validation accuracy is extremely low and doesn't change throughout training:
Final: val_loss: 12.7359 - val_acc: 0.0492

Validation accuracy is roughly 1/n where n is my number of classes.

When I manually evaluate my validation set, my results confirm the train accuracy to be mostly correct.

So my question is, has anyone experience with the validation accuracy being calculated incorrectly? Or does someone have an idea what may cause this.

**Comment by fatahfattah** [comment]
Issue was using two different ways of preprocessing for my imageGenerators.
